src/tiled/command.cpp

```diff
@@ -50,6 +51,34 @@ QString Command::finalCommand() const
     return finalCommand;
 }
 
+QString Command::finalWorkingDir() const
+{
+    QString finalWorkingDir;
+
+    MapDocument *mapDoc = DocumentManager::instance()->currentDocument();
+    if (mapDoc) {
+        const QString mapPath = mapDoc->path();
+        finalWorkingDir.replace(QLatin1String("%executablepath"),
+                                QLatin1String("\"%1\"")).arg(executablePath());
+        finalWorkingDir.replace(QLatin1String("%mappath"),
+                                QLatin1String("\"%1\"")).arg(mapPath);
+
+    }
+    return finalWorkingDir;
+}
+
+QString Command::executablePath() const
+{
+    QString finalcommand = finalCommand();
+
+    // removing parameters from command string
+    QString executableFilename(finalcommand.mid(0, finalcommand.lastIndexOf(QLatin1String(" ")) - 1));
```

Comment:
Parameters always should start after a space. So there are possible spaces in the path to executable itself and in those cases path should be surrounded with ' / ". Maybe next strategy won't be tricky?
See if the string contains ' / ". YES => use last " as executable path. NO => get substring until first space.
Also we can try to create some validator for string field to replace user's string to some parseable format. I don't sure too what way would be good here...